CMakeLists.txt

```diff
@@ -337,9 +336,13 @@ if (StbImage_FOUND)
         "multiple translation units!  Try upgrading your STB implementation, "
         "or using the auto-downloader (set DOWNLOAD_DEPENDENCIES=ON in the "
         "CMake configuration command.")
-  endif ()
+  endif()
 endif()
 
+if (STB_AVAILABLE)
+  add_definitions(-DMLPACK_USE_SYSTEM_STB)
```

Comment:
Can you add the `MLPACK_USE_SYSTEM_STB` option to `doc/user/compile.md`?  We should point out that we require `stb_image.h`, `stb_image_write.h`, and `stb_image_resize2.h` to be available (assuming you update to that version).